### ISSUE TITLE
Tests take 8+ hours on Python 3.5

### ISSUE THREAD

**Issue by jmwoloso** [Theano/Theano#4686]
Hello!

OS: Linux Mint 17.3
Python: 3.5 (Intel Distribution)
Theano: 8.2
Intel MKL

I built Theano from source on 3.4.3 and ran the tests and everything ran great in <30 mins. Now I've moved on and built it for Python 3.5 and the tests have been taking hours (like 8 hours). It seems to hang on <code>tests/test_extra_ops.py</code> or at least I think so since it is stuck showing a deprecation warning for the function on line 465. 

Here is the last few lines of where it is stuck. The cursor hasn't moved for hours from that spot.

```
/home/jwolosonovich/.virtualenvs/ztrip35i/lib/python3.5/site-packages/theano/tensor/tests/test_extra_ops.py:465: DeprecationWarning: This function is deprecated. Please call randint(1, 5 + 1) instead
  r = np.random.random_integers(5, size=(1,)).astype(dtype)
.......S......................
```

Any advice?

**Comment by zym1010** [comment]
On a CentOS 6.2 machine with Titan X and dual CPUs, I got the following testing time for Python 3.5+openblas+Theano 0.8.2

~~~
Ran 21747 tests in 9843.964s

OK (SKIP=85)
~~~

**Comment by nouiz** [comment]
To both of you, did you ran Theano tests before the timing or not? This could explain some time change. But I would expect more like the 9k-10k seconds

**Comment by zym1010** [comment]
@nouiz what do you mean by timing? and BTW, what's the typical testing time for Theano 0.8.2, with cuDNN v5, and default compiling mode (which is equivalent to `FAST_RUN`?)

**Comment by nouiz** [comment]
I'm not sure how long it should take. It depend of the computer/gpus, ... I was telling that the time you got 9843s is in what I would expect. But I could be wrong. Theano tests take times (it is a compiler, so we have many corner cases to tests).

If you change the default mode to DebugMode, then it get super slow, especially with cudnn. We did some improvement in test time with cudnn and DebugMode in the master of Theano.

 @jmwoloso can you start your run with the parameter -v, to know exactly which test seem to hang? Also, do you see it in top as working, or it seem to just wait and do nothing?

**Comment by zym1010** [comment]
@nouiz  yeah. I thought I did ran `theano.test()` when it was 0.7, and it took shorter time. Probably I guess it's because more and more tests are being added...

**Comment by nouiz** [comment]
Yes, as we have more features:)

**Comment by jmwoloso** [comment]
Hello @nouiz and @zym1010 

Sorry, haven't checked this for a couple days. I'm not using the GPU at all (I have an AMD GPU on my work laptop). Just the CPU and it's a Intel Core i7-6820HQ with 8GB of RAM so I wouldn't imagine that is the problem. The tests are relatively speedy when run on Python 3.4.3 as well. They just hang when I run them on the Intel Python 3.5 distribution (which could be the problem perhaps?)

The output I posted is the last of it that appears during testing. It just sits and hangs at whatever test it's running at that point in time. My `gcc` is 4.8 also. I used the import statement and then theano.test() to run them. Are you saying to use `theano-nose -v` instead, similar to the verbose output that `nosetests` provides?

Thank you both for your help as well!
